Public/Src/Tools/ContentPlacementAnalysisTools/ContentPlacementAnalysisTools.Extraction/Main/BuildDownloader.cs

```diff
@@ -0,0 +1,276 @@
+using System.Text;
+using System.Threading.Tasks.Dataflow;
+using BuildXL.ToolSupport;
+using ContentPlacementAnalysisTools.Core;
+using ContentPlacementAnalysisTools.Extraction.Action;
+using ContentPlamentAnalysisTools.Core;
+using Newtonsoft.Json;
+
+namespace ContentPlacementAnalysisTools.Extraction.Main
+{
+
+    /// <summary>
+    /// Main class for downloading a set of builds and getting sample artifacts from them
+    /// </summary>
+    public class BuildDownloader
+    {
+        private static readonly NLog.Logger s_logger = NLog.LogManager.GetCurrentClassLogger();
+
+        private static void Main(string[] args)
+        {
+            // parse args
+            var arguments = new Args(args);
+            if (arguments.Help)
+            {
+                return;
+            }
+            s_logger.Info($"BuildDownloader starting");
+            try
+            {
+                s_logger.Info($"Using configuration [{arguments.AppConfig}]");
+                // so in here we will create a new network.
+                var buildInfoBlock = new TransformManyBlock<GetKustoBuildInput, KustoBuild>(i =>
+                {
+                    var action = new GetKustoBuild(arguments.AppConfig);
+                    var result = action.PerformActionWithResult(i);
+                    // its not worth it to continue if this fails
+                    if (!result.ExecutionStatus)
+                    {
+                        s_logger.Error(result.Exception, "Could not download builds");
+                        throw result.Exception;
+                    }
+                    return result.Result.KustoBuildData;
+                });
+                var downloadBlock = new TransformBlock<KustoBuild, TimedActionResult<BuildDownloadOutput>>( i => 
+                {
+                    var action = new BuildDownload(arguments.AppConfig, arguments.OutputDirectory);
+                    return action.PerformActionWithResult(i);
+                }, 
+                new ExecutionDataflowBlockOptions()
+                {
+                    MaxDegreeOfParallelism = arguments.AppConfig.ConcurrencyConfig.MaxDownloadTasks
+                });
+                var decompressBlock = new TransformBlock<TimedActionResult<BuildDownloadOutput>, TimedActionResult<DecompressionOutput>>( i =>
+                {
+                    // check
+                    if (i.ExecutionStatus)
+                    {
+                        // decompress if build was successfull
+                        var action = new Decompression(arguments.AppConfig);
+                        return action.PerformActionWithResult(i.Result);
+                    }
+                    return new TimedActionResult<DecompressionOutput>(i.Exception);
+                },
+                new ExecutionDataflowBlockOptions()
+                {
+                    MaxDegreeOfParallelism = arguments.AppConfig.ConcurrencyConfig.MaxDecompressionTasks
+                });

```

Comment:
would you mind indenting this block one level?
that will make it a bit easier to read, I was initially thinking you were newing up ExecutionDataflowBlockOptions without assigning to a variable :)